Усечение строки

Red-Warrior opened the pull request.

Red-Warrior committed:
Усечение строки

jsru commented:
Поздравляю, все задачи решены абсолютно верно
```diff
Усечение строки
+ должна усекать строку 
+ должна не трогать короткую строку 
```

Добавляю преподавателя (@stas-dolgachov) для код-ревью, посмотрим что он скажет :)

dolgachio commented on 1-module/5-task/index.js:
```diff
@@ -5,5 +5,8 @@
 function truncate(str, maxlength) {
-  // ваш код...
+  if (str.length > maxlength) {
+    return str.slice(0, 19) + "…";
```
Отличное решение, в тоже время вы не могли бы использовать `maxlength` из условия, а не просто число 19? Предполагается, что ваше решение должно работать для любых входных значений, а не только тех, которые в тестах

Red-Warrior replied:
Да, конечно :)

Red-Warrior committed:
Использование maxlength из условия

jsru commented:
@stas-dolgachov ping :)

Red-Warrior committed:
Merge branch 'master' of https://github.com/js-tasks-ru/jsbasic-20200727_sword-and-brush

Red-Warrior committed:
Валидация имени пользователя.

Добавил ещё проверку на null, так как если в ответ на команду promt
нажать 'esc' , то это вызовет ошибку:
Uncaught TypeError: Cannot read property 'length' of null

jsru commented:
@stas-dolgachov ping :)

Red-Warrior commented:
Вот ещё вариант (Валидация имени пользователя):

function isValid(name) {
    let total = (name === null || name === "" || name.length < 4 || name.includes(" ")) ? false : true;
    return total;
}
